src/common/Throttle.cc

```diff
@@ -725,31 +766,44 @@ TokenBucketThrottle::~TokenBucketThrottle()
 
 void TokenBucketThrottle::set_max(uint64_t m) {
   std::lock_guard<Mutex> lock(m_lock);
-  m_throttle.set_max(m);
+  m_throttle.set_cbs(m);
 }
 
 void TokenBucketThrottle::set_average(uint64_t avg) {
-  m_avg = avg;
+  bool restart = false;
+  {
+    std::lock_guard<Mutex> lock(m_lock);
+    m_avg = avg;
+    if (m_throttle.set_cir(avg)) {
+      restart = true;
+      m_schedule_tick = m_throttle.tick;
+    }
+  }
+
+  if(restart) {
+    std::lock_guard<Mutex> timer_locker(*m_timer_lock);
+    cancel_timer();
+    schedule_timer();
```

Comment:
why restart?

Reply:
Because the filling period is not fixed. Different cir values correspond to different periods. I restart the schedule because the cir may be reset when the IO job is running.